DataStructures/SegmentTrees/SegmentTreeApply.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+
+namespace DataStructures.SegmentTrees
+{
+    /// <summary>
+    /// This is an extension of a segment tree, which allows to apply distributive operations to a subarray (in this case muliplication).
+    /// </summary>
+    public class SegmentTreeApply : SegmentTree
```

Comment:
```suggestion
    public class SegmentTreeApplyMultiplication : SegmentTree
```
IMO we can have other trees that apply different operations, so it's better to state what we apply here

Reply:
With this class operations can be applied to queries. If I would rename it, it sounds like the "main operation" (on which the tree is built) would be the multiplication and not the addition.

Reply:
It makes sense